server/service/src/sync/README.md

```diff
@@ -1,3 +1,99 @@
+
+## Record Types
+
+Records are treated differently by central server and remote sites based on data types, for example remote records belong to only one store and are only editable on a site where this store is active, whereas central data is only editable on a central server, the former usually syncs from remote site to central server whereas the later will only travel from central server to remote site.
+
+`remote record/data`: Can only be owned and edited by one store, on a site where this store is active, this data typically travels from remote site where it’s created and edited, to central server, which in turn may decide to forward it to another remote site (in case of transfers). Shipment, requisition, stock_line are examples of remote record.
+ 
+`central record/data`: This record is only editable on central server, and only travels in one direction, from central server to all remote sites. All central data will sync to all remote sites. Items and units are examples of central records.
+
+`central-remote record/data`: Similar in editability to central data, but will only sync to remote sites where these records are relevant. For example, master_list or name_store_join.
+
+`transfer record/data`: This is remote record but for a store that is not active on current site, this record is visible but not editable. For example, request requisition in response site.
+
+`shared-remote record/data`: Editable on any site and sync to all sites. For example barcodes
+
+`patient record/data`: Editable and accessible on any site where the patient is visible. Visibility in this case is determined by name store join. Prescribers, patients and patient documents are examples of patient record.
+
+
+## Phases/Stages of Synchronisation:
+
+`initialisation`: After the site is configured, it will first need to be initialised. This is where all relevant central and remote records are queued and sent to a remote site. Remote site is not ‘usable’ until initialisation stage is completed and all records are received
+
+`operational`: Once initialised the site will be in operational synchronisation phase, any changes to remote data will be sent to central server, and any changes to central data will travel on remote site from central server.
+
+We also consider central server as `backup service` for remote sites, in case of data loss at remote site, it can be re-initialised (usually by clearing app data, and switching site back to initialisation phase). Central server protects multiple instances of mSupply syncing for the same site by recording hardware id during each sync API request, if hardware id does not match previously stored hardware id an error will be returned.
```

Comment:
should it be 'open mSupply'?
```suggestion
We also consider central server as `backup service` for remote sites, in case of data loss at remote site, it can be re-initialised (usually by clearing app data, and switching site back to initialisation phase). Central server protects multiple instances of Open mSupply syncing for the same site by recording a device identifier ( hardware id ) during each sync API request, if the hardware id does not match the previously stored hardware id then an error will be returned.
```

Reply:
> should it be 'open mSupply'?

It's actually for both, but I guess Open mSupply makes it less confusing, since we haven't mentioned mSupply anywhere in the docs as remote site, thanks